source/dem/dem.cc

```diff
@@ -1115,6 +1115,17 @@ DEMSolver<dim>::solve()
             triangulation,
             parameters.boundary_conditions);
 
+  // Store information about floating mesh/background mesh intersection
+  for (unsigned int i_solid = 0; i_solid < solids.size(); ++i_solid)
+    {
+      floating_mesh_info[i_solid] =
+        solids[i_solid]->map_solid_in_background_triangulation(triangulation);
+    }
+
+  // Set insertion object type before the restart. We need to do so, otherwise
+  // we don't know which kind of insertion in beging used.
+  insertion_object = set_insertion_type(parameters);
```

Comment:
```suggestion
  // Set insertion object type before the restart because the restart only rebuilds the member of the insertion object
  insertion_object = set_insertion_type(parameters);
```